monkey/infection_monkey/monkey.py

```diff
@@ -79,12 +79,24 @@ def initialize(self):
         self._network = NetworkScanner()
         self._dropper_path = sys.argv[0]
 
+        server_num=0     
+        secondary_servers= len(self._args) 
+  
         if self._default_server:
             if self._default_server not in WormConfiguration.command_servers:
                 LOG.debug("Added default server: %s" % self._default_server)
-                WormConfiguration.command_servers.insert(0, self._default_server)
+                WormConfiguration.command_servers.insert(server_num, self._default_server)
             else:
                 LOG.debug("Default server: %s is already in command servers list" % self._default_server)
+            server_num=server_num+1
+        
+        while server_num<(secondary_servers+1):
+            if self._args[server_num -1] not in WormConfiguration.command_servers:
+                LOG.debug("Added default server: %s" % self._args[server_num -1])
+                WormConfiguration.command_servers.insert(server_num, self._args[server_num-1])
+            else:
+                LOG.debug("Default server: %s is already in command servers list" % self._args[server_num-1])
```

Comment:
Ok. So you add server IPs to command servers list. Can you verify that monkey tries all IPs in that list when connecting to server?